go/tools/builders/asm.go

```diff
@@ -137,9 +140,12 @@ func buildSymabisFile(goenv *env, sFiles, hFiles []fileInfo, asmhdr string) (str
 	return symabisName, err
 }
 
-func asmFile(goenv *env, srcPath string, asmFlags []string, outPath string) error {
+func asmFile(goenv *env, srcPath, packagePath string, asmFlags []string, outPath string) error {
 	args := goenv.goTool("asm")
 	args = append(args, asmFlags...)
+	if packagePath != "" {
+		args = append(args, "-p", packagePath)
+	}
```

Comment:
This would make rules_go incompatible with lower version of go sdk (before 1.19) because `go tool asm` have yet to added this flag.

Do we need to do this in a separate file dedicated to 1.19?

Reply:
Good catch. I pushed a new commit where I only add the flag with Go 1.19+. For some reason, I do get duplicate definition build errors though - I'm probably using `//go:build` wrong. Do you know how I can fix it?